hasprefix.go

```diff
@@ -0,0 +1,40 @@
+package vfs
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// A Stater implements Stat.
+type Stater interface {
+	Stat(string) (os.FileInfo, error)
+}
+
+// HasPrefix returns true if p or any parent of p is the same file as prefix.
+// prefix must exist, but p may not. It is an expensive but accurate alternative
+// to the deprecated filepath.HasPrefix.
+func HasPrefix(fs Stater, p, prefix string) (bool, error) {
```

Comment:
IMO "HasPrefix" is a bit misleading. To me, it suggests that you're operating on paths, but you're actually operating on actual FS datastructures. "IsIn" or "Contains" or something like that would be more accurate.

Reply:
Good point. Renamed to `Contains`.